lib/rules/boolean-prop-naming.js

```diff
@@ -228,6 +228,46 @@ module.exports = {
       args.filter((arg) => arg.type === 'ObjectExpression').forEach((object) => validatePropNaming(node, object.properties));
     }
 
+    function getTypeAnnotation(component) {
+      // If this is a functional component that uses a global type, check it
+      if ((
+        component.node.type === 'FunctionDeclaration'
+        || component.node.type === 'ArrowFunctionExpression'
+      )
+        && component.node.params
+        && component.node.params.length
+        && component.node.params[0].typeAnnotation) {
+        return component.node.params[0].typeAnnotation.typeAnnotation;
+      }
+
+      if (
+        component.node.parent
+        && component.node.parent.type === 'VariableDeclarator'
+        && component.node.parent.id
+        && component.node.parent.id.type === 'Identifier'
+        && component.node.parent.id.typeAnnotation
+        && component.node.parent.id.typeAnnotation.typeAnnotation
+        && component.node.parent.id.typeAnnotation.typeAnnotation.typeParameters
+        && component.node.parent.id.typeAnnotation.typeAnnotation.typeParameters.type === 'TSTypeParameterInstantiation'
+      ) {
+        return component.node.parent.id.typeAnnotation.typeAnnotation.typeParameters.params.find(
+          (param) => param.type === 'TSTypeReference'
+        );
+      }
+    }
+
+    function findTypeAnnotations(identifier, node) {
+      if (node.type === 'TSTypeLiteral') {
+        const currentNode = objectTypeAnnotations.get(identifier.name) || [];
+        currentNode.push(node);
```

Comment:
```suggestion
        const currentNode = [].concat(objectTypeAnnotations.get(identifier.name) || [], node);
```
this way we avoid mutation